Services/Simulation/JavascriptInterpreter.cs

```diff
@@ -100,27 +82,33 @@ public class JavascriptInterpreter : IJavascriptInterpreter
             try
             {
                 Program program;
+                bool isInStorage = string.Equals(script.Path.Trim(),
+                    DeviceModelScript.DeviceModelScriptPath.Storage.ToString(),
+                    StringComparison.OrdinalIgnoreCase);
+                string filename = isInStorage ? script.Id : script.Path;
+
                 if (programs.ContainsKey(filename))
                 {
                     program = programs[filename];
                 }
                 else
                 {
-                    var sourceCode = this.LoadScript(filename);
+                    var task = this.LoadScriptAsync(filename, isInStorage);
+                    task.Wait(TimeSpan.FromSeconds(30));
```

Comment:
we need to remove the use of `.Wait`. I've tried doing it and the change propagates through the code to a point where there is a `lock` statement in `UpdateDeviceState`, so it's not a trivial change.

Please add a TODO here and a link to a work item to address the issue.  What we'll probably end up doing is loading all the scripts before the simulation starts, running all the asynchronous part at the start. Then we can keep the scripts in memory and this method will not need to be asynchronous, and load the script source code from memory.

Reply:
Done. Issue opened: https://github.com/Azure/device-simulation-dotnet/issues/240